File: Backend/Soft-Gestion-de-clientes/src/main/java/com/SoftGestionClientes/Model/Payment.java
package com.SoftGestionClientes.Model;

import com.SoftGestionClientes.Enums.EPaymentMethod;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @CreationTimestamp
    @Column(name = "payment_date")
    private LocalDate paymentDate;

    @ManyToOne
    @JoinColumn(name = "client_id")
    private Client client;

    @ManyToOne
    @JoinColumn(name = "report_id")
    private Report report;

    @NonNull
    private Double amount;

    @NonNull
    private EPaymentMethod paymentMethod;

    @OneToOne
    @JoinColumn(name = "sale_id")
    private Sale sale;
}

File: Backend-Node/models/payment.ts
import mongoose, { Document, Schema, model } from "mongoose";
import { EPaymentMethod } from "../enums/EPaymentMethod";
import { formatDateIso } from "../utils/dateUtils";

const paymentSchema = new Schema({
    payment_date: {type: String, default: formatDateIso(new Date())},
    clientId: {type: mongoose.Schema.Types.ObjectId, ref: "Client", required: true}, 
    amount: {type: Number, min: 0.01, required: true},
    payment_method: {type: String, enum: EPaymentMethod, required: true},
    saleId: {type: mongoose.Schema.Types.ObjectId, ref: "Sale"}, 
    reportId: {type: mongoose.Schema.Types.ObjectId, ref: "Report"}
})

const PaymentModel = model("Payment", paymentSchema, "payments")


export interface PaymentDocument extends Document{
    _id: mongoose.Types.ObjectId,
    payment_date: string;
    clientId: mongoose.Types.ObjectId;
    amount: number;
    payment_method: string;
    saleId?: mongoose.Types.ObjectId | undefined;
    reportId?: mongoose.Types.ObjectId | undefined;
  }

export default PaymentModel